### ISSUE TITLE
Clientlib js versioning is not working after upgrading to SP8 even with ACS Commons Package v5.0.6

### ISSUE THREAD

**Issue by shalaka-scuk** [Adobe-Consulting-Services/acs-aem-commons#2664]
## Required Information

* [ ] AEM Version, including Service Packs, Cumulative Fix Packs, etc: _AEM 6.5.8______
* [ ] ACS AEM Commons Version: __5.0.6___
* [ ] Reproducible on Latest? yes/no Yes

**### Expected Behavior**

A has should be generated for the versioning of AEM Client lib Js and Css files using the acs-aem-commons packge

**### Actual Behavior**

Our clientlib js versioning has stopped working after upgrading to AEM SP8 on 6.5.

We have seen this issue raised by others already and we followed the suggestion to install the latest version of ACS Commons Package, which is v5.0.6

We installed the ACS Commons Package v5.0.6 from https://javalibs.com/artifact/com.adobe.acs/acs-aem-commons-content and installed on our AEM using package manager.

In our POM we have:
<dependency>
			<groupId>com.adobe.acs</groupId>
			<artifactId>acs-aem-commons-bundle</artifactId>
			<version>5.0.6</version>
			<scope>provided</scope>
</dependency>

**### Steps to Reproduce**

Upgrade to Service Pack 8 on AEM 6.5
Get the latest version of ACS Commons Package v5.0.6 and install it on AEM from package manager

**### Links**

ACS Commons Package installed: https://javalibs.com/artifact/com.adobe.acs/acs-aem-commons-content
Our Website where the js files in the source are not hashed: www.savethechildren.org.uk

**Comment by shalaka-scuk** [comment]
Hello, Can someone please check and respond on this?

**Comment by YegorKozlov** [comment]
Clientlib versioning is working fine on my local AEM 6.5 SP9+ ACS Commons 5.0.6

/apps/weretail/config/rewriter/versioned-clientlibs.tidy.json
```
{
  "jcr:primaryType": "nt:unstructured",
  "enabled": true,
  "serializerType": "htmlwriter",
  "transformerTypes": [
    "linkchecker",
    "versioned-clientlibs"
  ],
  "order": "-1",
  "contentTypes": ["text/html"],
  "generatorType": "htmlparser"
  }
```

**Comment by shalaka-scuk** [comment]
Hi, 

We seem to have the same properties in our versioned-clientlibs folder (as seen in crx) which is under <our-project>/config/rewriter. We don't have a json under it. Do we need to add it? We haven't changed anything here.

Any suggestions on how to proceed?

**Comment by YegorKozlov** [comment]
.json is just for demo purposes. You don't need it. 

Do you have conflicting rewriter configurations? 
How many records does this query return? 
```
SELECT * FROM [nt:base] AS s WHERE ISDESCENDANTNODE([/apps]) AND NAME(s)='rewriter'
```

**Comment by shalaka-scuk** [comment]
We don't seem to have any conflicting rewriter configurations.

The given query returns only one path, which is:
/apps/our-project/config/rewriter

**Comment by YegorKozlov** [comment]
Try enabling debug logging on com.adobe.acs.commons.rewriter.* 

Do you see any messages / exceptions ? 
Try restarting your AEM instance.

**Comment by shalaka-scuk** [comment]
We have restarted the AEM instance multiple times. That dint help.
I will try to see if I can find any relevant exceptions

**Comment by shalaka-scuk** [comment]
Update we were able to see the following exception in our Publish logs:

Caused by: org.apache.sling.api.SlingException: javax.servlet.ServletException: java.lang.NoClassDefFoundError: com/adobe/acs/commons/errorpagehandler/ErrorPageHandlerService
	at org.apache.sling.scripting.jsp.jasper.servlet.JspServletWrapper.handleJspExceptionInternal(JspServletWrapper.java:691)
	at org.apache.sling.scripting.jsp.jasper.servlet.JspServletWrapper.handleJspException(JspServletWrapper.java:608)
	at org.apache.sling.scripting.jsp.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:525)
	at org.apache.sling.scripting.jsp.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:449)
	at org.apache.sling.scripting.jsp.JspScriptEngineFactory.callJsp(JspScriptEngineFactory.java:339)
	at org.apache.sling.scripting.jsp.JspScriptEngineFactory.access$100(JspScriptEngineFactory.java:97)
	at org.apache.sling.scripting.jsp.JspScriptEngineFactory$JspScriptEngine.eval(JspScriptEngineFactory.java:600)
	at org.apache.sling.scripting.core.impl.DefaultSlingScript.call(DefaultSlingScript.java:388)
	... 79 common frames omitted
Caused by: java.lang.NoClassDefFoundError: com/adobe/acs/commons/errorpagehandler/ErrorPageHandlerService
	at org.apache.jsp.apps.sling.servlet.errorhandler.default_jsp._jspService(default_jsp.java:310)
	at org.apache.sling.scripting.jsp.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:725)
	at org.apache.sling.scripting.jsp.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:502)
	... 84 common frames omitted
ERROR 2021-08-13 15:53:18,955; logger=o.a.s.s.r.i.SlingServletResolver;  message=Original error class java.io.IOException
java.io.IOException: Unable to get component of class 'interface org.apache.sling.rewriter.Transformer' with type 'versioned-clientlibs'.
	at org.apache.sling.rewriter.impl.PipelineImpl.getPipelineComponent(PipelineImpl.java:161)
	at org.apache.sling.rewriter.impl.PipelineImpl.init(PipelineImpl.java:103)
	at org.apache.sling.rewriter.impl.ProcessorManagerImpl.getProcessor(ProcessorManagerImpl.java:480)
	at org.apache.sling.rewriter.impl.RewriterResponse.getProcessor(RewriterResponse.java:172)
	at org.apache.sling.rewriter.impl.RewriterResponse.getWriter(RewriterResponse.java:110)
	at org.apache.sling.scripting.core.impl.helper.OnDemandWriter.getWriter(OnDemandWriter.java:38)
	at org.apache.sling.scripting.core.impl.helper.OnDemandWriter.write(OnDemandWriter.java:61)
	at java.io.PrintWriter.write(PrintWriter.java:426)
	at org.apache.sling.scripting.core.impl.helper.OnDemandWriter.write(OnDemandWriter.java:61)
	at java.io.PrintWriter.write(PrintWriter.java:426)

**Comment by davidjgonzalez** [comment]
Can you ensure that the ACS AEM commons osgi bundle has started?

If it has, please try recompiling your sling scripts.

It sounds like the bundle hasn't started and the service interface isn't being exported, causing the jsp to be unable yo resolve the class

**Comment by shalaka-scuk** [comment]
From what I can see, the ACS AEM commons bundle has started. I check under <IP>/system/console/status-Bundles and I can see:
Bundle 530 - ACS AEM Commons Bundle 5.0.6 (state: Active)
    Symbolic Name: com.adobe.acs.acs-aem-commons-bundle
    Version: 5.0.6
    Bundle Location: jcrinstall:/apps/acs-commons/install/acs-aem-commons-bundle-3.14.8.jar

Also under <IP>/system/console/bundles, we have:



Symbolic Name | com.adobe.acs.acs-aem-commons-bundle
-- | --
Version | 5.0.6
Bundle Location | jcrinstall:/apps/acs-commons/install/acs-aem-commons-bundle-3.14.8.jar
Last Modification | Mon Aug 02 12:08:09 UTC 2021
Bundle Documentation | https://adobe-consulting-services.github.io/acs-aem-commons/
Vendor | Adobe
Description | Main ACS AEM Commons OSGi Bundle. Includes commons utilities.
Start Level | 20

As per your instructions I also went to IP/libs/granite/ui/content/dumplibs.rebuild.html?back=true
and did "Invalidate Caches" followed by 'Rebuild Libraries"... We face the same issue and continue to see the same error in the logs even after these steps.
Anything else I can try. Is there any other step I am missing for recompiling sling scripts?

**Comment by shalaka-scuk** [comment]
Note that we first installed this package on the Author (Not on publish instances), until then it does not give any errors in any logs, but the client-libs versioning does not work. But once we replicated the package to the publish instances it started given the above mentioned errors in the logs and some of our important pages stopped loading with a 503 error.

Now my question is, Does this package need to be replicated to the Publish instance as well? Or is it enough to install it on the Author?

**Comment by YegorKozlov** [comment]
Did you have a previous installation of ACS Commons? The bundle path is weird: ```jcrinstall:/apps/acs-commons/install/acs-aem-commons-bundle-3.14.8.jar ```

It should be
```
jcrinstall:/apps/acs-commons/install/acs-aem-commons-bundle-5.0.6.jar
```
Try to un-install the previous version of ACS  Commons and then install 5.0.6.

**Comment by shalaka-scuk** [comment]
Yes, we did have a previous version of ACS but that was v4.0.3 .... in the system bundles in AEM, it always shows a weird version in the name for most bundles i think. When we installed the new version 5.0.6 the old version, the older version automatically got removed somehow. 
So if I go to the package manager now, and search for ACS I only have the new version. Dont have the older version anymore so there is not way to uninstall it
